src/axl_io.c

```diff
@@ -408,8 +606,17 @@ int axl_file_copy(const char* src_file, const char* dst_file, unsigned long buf_
         rc = AXL_FAILURE;
     }
 
-    /* unlink the file if the copy failed */
-    if (rc != AXL_SUCCESS) {
+    if (rc == AXL_SUCCESS) {
+        /* succeeded in copying the file, now copy uid/gid,
+         * permissions, and timestamps */
+        if (axl_copy_metadata) {
+            kvtree* meta = kvtree_new();
+            axl_meta_encode(src_file, meta);
+            axl_meta_apply(dst_file, meta);
```

Comment:
Do you really need to encode the values into a KVTree?  Why not just pass `sb` and `nsecs` to `axl_meta_apply()`?

Reply:
The main reason is code reuse.  These same meta_encode/meta_apply functions are also in the shuffile and redset components, so they are duplicated across three components now.  I'm hoping we can extract this logic into a new I/O component as future work, and then we can replace all of these calls to use the new common I/O component.

Reply:
Oh, and forgot to add, they are encoded to kvtrees in those other components because the info has to be saved to files and transferred across nodes with MPI.  The kvtrees make that easy to do.